### DISCUSSION TITLE
Tools for Memory Profiling UNO Platform iOS application

### DISCUSSION THREAD

**Discussion by dtrovato997** [unoplatform/uno#17568] [Q&A]
Hello,

I have an UNO Platform application which runs on iOS UNO, it uses UNO 5.2, the framework for iOS is .NET8-iOS.

I've experienced crashes which produced jetsam logs which pointed out that the app used too much memory(per-process-limit error).

So I decided I wanted to start to profile the application for memory usage to find out why it uses too much memory

I've done a research and i'm not completely sure what I should do to address this problem:

I tried the Allocations tool of XCode, but this crashes the applicatiom after 30 second or so, before investigating further I wanted to ask, is it possible to profile UNO Platform iOS App with this tool in the first place?

I've read about the Memory Instance Counter : https://platform.uno/docs/articles/uno-development/troubleshooting-memory-issues.html, is this reliable for iOS? What are the limitation of this tools?

Is it possible in uno platform to use tools like dotnet-gcdump like it is suggested in this blog for .NET-MAUI? https://github.com/dotnet/maui/wiki/Memory-Leaks or is this something exclusive to .NET MAUI?

In general I want to ask, what tool can i use to address this problem in UNO Platform for iOS?

**Comment by jeromelaban** [comment]
Thanks for the question. Uno Platform uses .NET, and most of the tooling that is provided, like `dotnet-gcdump` or `dotnet-trace` does work properly.

You can also [profile the CPU usage](https://platform.uno/docs/articles/guides/profiling-applications.html#profiling-net-ios-applications).
